### DISCUSSION TITLE
Checked exception when sending message using PulsarTemplate

### DISCUSSION THREAD

**Discussion by jonas-grgt** [spring-projects/spring-pulsar#546] [General]
It feels unnecessary to force the users of `PulsarTemplate` to have to deal with a checked `PulsarClientException`. 
Other spring template implementation do not force this on their clients which makes their code in the end much readable.
Why was this option chosen?

**Comment by onobc** [comment]
I don't quite remember why we chose the checked exception but I do agree w/ you @jonasgeiregat and would be happy to remove it in the 1.1.0 release (1.1.0-M1 due next month).

**Comment by onobc** [comment]
Closing via https://github.com/spring-projects/spring-pulsar/issues/547
